python/modules/IcePy/Current.cpp

```diff
@@ -7,313 +7,95 @@
 #include "Ice/ObjectAdapter.h"
 #include "ObjectAdapter.h"
 #include "Util.h"
-#include <structmember.h>
 
 using namespace std;
 using namespace IcePy;
 
-namespace IcePy
+PyObject* IcePy::createCurrent(const Ice::Current& current)
 {
-    struct CurrentObject
-    {
-        PyObject_HEAD Ice::Current* current;
-        PyObject* adapter;
-        PyObject* con;
-        PyObject* id;
-        PyObject* facet;
-        PyObject* operation;
-        PyObject* mode;
-        PyObject* ctx;
-        PyObject* requestId;
-        PyObject* encoding;
-    };
+    PyObject* currentType = lookupType("Ice.Current");
 
-    //
-    // Member identifiers.
-    //
-    const Py_ssize_t CURRENT_ADAPTER = 0;
-    const Py_ssize_t CURRENT_CONNECTION = 1;
-    const Py_ssize_t CURRENT_ID = 2;
-    const Py_ssize_t CURRENT_FACET = 3;
-    const Py_ssize_t CURRENT_OPERATION = 4;
-    const Py_ssize_t CURRENT_MODE = 5;
-    const Py_ssize_t CURRENT_CTX = 6;
-    const Py_ssize_t CURRENT_REQUEST_ID = 7;
-    const Py_ssize_t CURRENT_ENCODING = 8;
-}
+    PyObjectHandle args = PyTuple_New(9);
 
-extern "C" CurrentObject*
-currentNew(PyTypeObject* type, PyObject* /*args*/, PyObject* /*kwds*/)
-{
-    CurrentObject* self = reinterpret_cast<CurrentObject*>(type->tp_alloc(type, 0));
-    if (!self)
+    PyObject* adapter = wrapObjectAdapter(current.adapter);
```

Comment:
Does this create a new PyObject* each time, or do we cache and reuse this object?

Reply:
It is currently creating a new one, which may not be correct.

Reply:
wrapObjectAdapter calls createObjectAdapter and the ObjectAdapter python wrapper is not cached. it also creates a C++ mutex that is used for calling hold. It might be worth using a cache like we do for the communicator.